internal/component/otelcol/exporter/exporter.go

```diff
@@ -92,7 +100,7 @@ type Exporter struct {
 
 	// Signals which the exporter is able to export.
 	// Can be logs, metrics, traces or any combination of them.
-	supportedSignals TypeSignal
+	supportedSignals TypeSignalFunc
```

Comment:
```suggestion
	// This is a function because which signals are supported may depend on the component configuration.
	supportedSignals TypeSignalFunc
```